core/classes/Core/Util.php

```diff
@@ -128,41 +129,140 @@ public static function isExternalURL(string $url): bool {
         return !(str_replace('www.', '', rtrim(self::getSelfURL(false), '/')) == str_replace('www.', '', $parsed['host']));
     }
 
+    /**
+     * @return array List of trusted proxy networks according to config file and environment
+     */
+    private static function getTrustedProxies(): array {
+        $trustedProxies = [];
+
+        // Add trusted proxies from config file
+        $configProxies = Config::get('core/trustedProxies');
+        if ($configProxies !== false) {
+            if (!is_array($configProxies)) {
+                die('Trusted proxies should be an array');
+            }
+            $trustedProxies = array_merge($trustedProxies, $configProxies);
+        }
+
+        // Add trusted proxies from environment variable (comma-separated string)
+        $envProxies = getenv('NAMELESS_TRUSTED_PROXIES');
+        if ($envProxies !== false) {
+            $envProxiesArray = explode(',', $envProxies);
+            $trustedProxies = array_merge($trustedProxies, $envProxiesArray);
+        }
+
+        return $trustedProxies;
+    }
+
+    /**
+     * Checks whether the client making the request is a trusted proxy. If not,
+     * abruptly aborts the request using die().
+     */
+    private static function ensureTrustedProxy(): void {
+        $trustedProxies = self::getTrustedProxies();
+
+        if (count($trustedProxies) === 0) {
+            die('Received proxy header but no trusted proxies are configured. Please see <a href="https://docs.namelessmc.com/trusted-proxies">https://docs.namelessmc.com/trusted-proxies</a> for more information.');
+        }
+
+        $trusted = false;
+
+        foreach ($trustedProxies as $trustedProxy) {
+            if (IpUtils::checkIp($_SERVER['REMOTE_ADDR'], $trustedProxy)) {
```

Comment:
I'm probably missing something here, but shouldn't we be using `self::getRemoteAddress()` here, instead of `$_SERVER['remote_address']`? (I know that `getRemoteAddress` will call `ensureTrustedProxy`, but we could add a param to skip the trusted proxy check)